src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -8,6 +8,7 @@
 * (Deliberately contains bugs.)
 */
 public class TorpedoStore {
+  private final Random random = new Random();
```

Comment:
This is an optimal solution to speed up method calls and put less work on the garbage collector.